ide-support/revsaveasandroidstandalone.livecodescript

```diff
@@ -1907,7 +1907,7 @@ private function EncodeString pString, pInEncoding, pOutEncoding
       put textDecode(pString, pInEncoding) into pString
    end if
    
-   if pOutEncoding is not empty then
+   if pOutEncoding is not empty and not (pString contains "android:label") then
```

Comment:
@livecodepanos are we sure this is necessary. From the looks of things we are writing the template manifest as utf8 then reading it in again and decoding as utf8 in order to merge so when we re-encode the merged manifest (which I think is here) we would not want to skip it I think.

Reply:
@montegoulding I am not 100% sure what happens here, but I noticed that if I remove this, then the app label (say `样品`) appears as `Ê†∑ÂìÅ` (which is the `textEncode("样品","utf-8")` ) in the device.

Reply:
hmm... is whatever you are viewing it in reading it as utf8?

Reply:
oh... the app label... hmm

Reply:
@livecodepanos I have found the problem with the encoding here. The issue is the call to ConvertArrayToXML does not pass in the encoding of the array and therefore it does not do the decode before encoding. https://github.com/livecodepanos/livecode/blob/f6fc285f56973023573c426ff589182dd0f3ed10/ide-support/revsaveasandroidstandalone.livecodescript#L1602

Reply:
So I'm fairly sure if you change that line to `ConvertArrayToXML(tMainArrayA, "utf-8")` then you won't need to have this special case in `EncodeString` because it will do the `textDecode`... BTW it might be worthwhile adding this to the top of `EncodeString` so we aren't wasting our time:
```
if pInEncoding is pOutEncoding then
     return pString
end if
```